webui/react/src/components/Bar.tsx

```diff
@@ -0,0 +1,53 @@
+import { Tooltip } from 'antd';
+import React from 'react';
+
+import { ShirtSize } from 'themes';
+import { floatToPercent } from 'utils/string';
+
+import css from './Bar.module.scss';
+
+export interface BarPart {
+  percent: number; // between 0-1
+  color: string; // css color
+  label?: string;
+}
+
+export interface Props {
+  barOnly?: boolean;
+  parts: BarPart[];
+  size?: ShirtSize;
+}
+
+const partStyle = (part: BarPart) => {
+  return {
+    backgroundColor: part.color,
+    width: floatToPercent(part.percent, 0),
+  };
+};
+
+const Bar: React.FC<Props> = ({ barOnly, parts, size }: Props) => {
+  const classes: string[] = [ css.base ];
+
+  if (barOnly) classes.push(css.barOnly);
+
+  return (
+    <div className={classes.join(' ')}>
+      <div
+        className={css.bar}
+        style={{ height: `var(--theme-sizes-layout-${size || ShirtSize.tiny})` }}>
+        <ol>
+          {parts.filter(part => part.percent !== 0 && !isNaN(part.percent)).map((part, idx) => {
+            return (
+              <Tooltip key={idx} title={part.label}>
+                <li style={partStyle(part)} />
+              </Tooltip>
+            );
+          })}
+        </ol>
```

Comment:
**suggestion:** `<li>` is typically designed to sit flatly inside an `ol` or `ul`, but the wrapping it with a tooltip `div` breaks this pattern. browser might render this ok, but it might just be better to use `div` instead.